cmd/sink/expr/cel.go

```diff
@@ -0,0 +1,211 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package expr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	celext "github.com/google/cel-go/ext"
+	"github.com/kubearchive/kubearchive/cmd/operator/api/v1alpha1"
+	"github.com/kubearchive/kubearchive/pkg/files"
+	"gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const globalKey = "_global"
+
+type Filters struct {
+	archive map[string]cel.Program
+	delete  map[string]cel.Program
+}
+
+// Returns a Filters struct with an empty archive and delete slice
+func EmptyFilters() *Filters {
+	return &Filters{
+		archive: make(map[string]cel.Program),
+		delete:  make(map[string]cel.Program),
+	}
+}
+
+// Creates a Filters struct from the path to a directory where a ConfigMap was mounted. If path is empty string or path
+// does not exist, it returns a Filters struct with empty archive and delete slices. It will attempt to create all the
+// cel programs that it can from the ConfigMap. Any errors that are encountered are wrapped together and returned. Even
+// if this function returns an error, the Filters struct returned can still be used and will not be nil.
+func NewFilters(path string) (*Filters, error) {
+	var errList []error
+	filters := &Filters{
+		archive: make(map[string]cel.Program),
+		delete:  make(map[string]cel.Program),
+	}
+	exists, err := files.PathExists(path)
+	if err != nil {
+		return filters, fmt.Errorf("cannot determine if ConfigMap is mounted at path %s: %s", path, err)
+	}
+	if path == "" || !exists {
+		return filters, fmt.Errorf("cannot create Filters. ConfigMap is not mounted at path %s", path)
+	}
+	filterFiles, err := files.DirectoryFiles(path)
+	if err != nil {
+		return filters, fmt.Errorf(
+			"cannot create Filters. Could not read files created from ConfigMap mounted at path %s: %s",
+			path,
+			err,
+		)
+	}
+	for namespace, filePath := range filterFiles {
+		resourcesBytes, err := os.ReadFile(filePath)
+		if err != nil {
+			errList = append(errList, err)
+			continue
+		}
+		resources := []v1alpha1.KubeArchiveConfigResource{}
+		err = yaml.Unmarshal(resourcesBytes, &resources)
+		if err != nil {
+			errList = append(errList, err)
+			continue
+		}
+		for _, resource := range resources {
+			namespaceGvk := NamespaceGroupVersionKind(namespace, resource)
+			archiveExpr, err := CreateCelExpr(resource.ArchiveWhen)
+			if err != nil {
+				errList = append(errList, err)
+			} else {
+				filters.archive[namespaceGvk] = *archiveExpr
+			}
+			deleteExpr, err := CreateCelExpr(resource.DeleteWhen)
+			if err != nil {
+				errList = append(errList, err)
+			} else {
+				filters.delete[namespaceGvk] = *deleteExpr
+			}
+		}
+	}
+
+	return filters, errors.Join(errList...)
+
+}
```

Comment:
You probably should not have lines in a method after the `return`, even if they are empty.

```suggestion
	return filters, errors.Join(errList...)
}
```

Reply:
fixed